[#5] 커스텀 로그인 및 소셜 로그인 구현

codingbaraGo opened the pull request:
<!-- 제목은 `[#이슈번호] 제목`으로 작성해주세요. (ex) [#1] 프로젝트 초기 설정 작업 -->

###🧑‍💻 작업 내용
- 커스텀 로그인 구현
- 소셜 로그인 (네이버) 구현
- .yml 파일들에 있는 민감한 정보들 삭제
- .env파일 추가


### ✨ 리뷰 포인트
<!-- (ex) query 가 너무 복잡한 것 같은데 이 위주로 봐주세요. -->
### 🎯 관련 이슈
<!-- pr이 merge 되면 이슈가 자동으로 close 되도록 합니다. 만약 자동 close 를 하지 않고 이슈만 링크한다면 closed 키워드를 삭제해주세요.-->
closed #5

w3schools-test committed:
feat: 커스텀 로그인 및 소셜 로그인 구현

w3schools-test committed:
fix: User.Entity.UserEntity와 User.Model.User를 병합

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/model/User.java:
```diff
@@ -1,50 +1,57 @@
-		this.puzId = puzId;
-		this.puzColor = puzColor;
-	}
+@Setter @Getter
```
setter 혹시 사용하시나용? 사용하지 않는다면 지워주세요!!

codingbaraGo replied:
여기저기서 끌어다 쓰고있네요...
builder만 쓰도록 수정해야겠어요

hikarigin99 commented on src/main/resources/application-dev.yml:
```diff
@@ -1,8 +1,8 @@
-    password: mema1234
+    url: jdbc:mysql://${DB_DEV_IP}/dev_db?useSSL=false&serverTimezone=UTC
+    username: ${DB_DEV_USERNAME}
+    password: ${DB_DEV_PASSWORD}
```
👍

hikarigin99 commented on src/main/java/com/swyp/mema/global/config/security/SecurityConfig.java:
```diff
@@ -0,0 +1,118 @@
+    private final CustomSuccessHandlerCookie customSuccessHandlerCookie;
+
+    public SecurityConfig(AuthenticationConfiguration authenticationConfiguration, JWTUtil jwtUtil, CustomOAuthUserService customOAuthUserService, CustomSuccessHandlerCookie customSuccessHandlerCookie) {
+        System.out.println("Security Config create");
```
@RequiredArgsConstructor 로 바꾸시는 거 어떠신가용

hikarigin99 commented on src/main/java/com/swyp/mema/global/security/jwt/filter/JWTFilter.java:
```diff
@@ -0,0 +1,83 @@
+    @Override
+    protected void doFilterInternal(HttpServletRequest  request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+        System.out.println("jwtfilter - do filter internal");
```
저희 `Lombok` 사용하기 때문에 `@Slf4j` 어노테이션 사용해서 log 찍는 방식이 좋을 것 같아용
언제 한번 날잡아서 sout 없애봐야겠네여🤔

codingbaraGo replied:
분명 쓰는법 강의에서 본거같은데
급하니까 그냥 sout을 찍어버렸네요ㅋㅋㅋㅋ
쓰는법 보고 수정해볼게요

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/dto/UserDTO.java:
```diff
@@ -0,0 +1,31 @@
+import lombok.Setter;
+
+@Getter @Setter
+public class UserDTO {
```
저희 네이밍 컨밴션 request 패키지 하위에 userReq 쓰기로 했습니당😭

codingbaraGo replied:
반영 완료입니다!

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/controller/JoinController.java:
```diff
@@ -0,0 +1,27 @@
+    private final JoinService joinService;
+
+    @PostMapping("/join/custom")
+    public ResponseEntity<String> joinCustom(@RequestBody UserDTO userDTO) {
```
DTO 에 `@NotBlank` 같은 어노테이션 쓰셨던데!!
`@Valid` 어노테이션 추가해서 검증해보시는건 어떠신가용🤔

codingbaraGo replied:
@NotBlank를 써놓고 활용을 안했네요
추가했습니다!

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/dto/converter/UserConverter.java:
```diff
@@ -0,0 +1,48 @@
+import com.swyp.mema.domain.user.dto.UserDTO;
+import com.swyp.mema.domain.user.model.User;
+
+public class UserConverter {
```
Converter 로 네이밍 변경하신거 굳입니다👍

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/service/JoinService.java:
```diff
@@ -0,0 +1,45 @@
+
+            /*
+                이 부분 exception 던지는 걸로 바꾸기
+             */
```
😳

codingbaraGo replied:
이것도 은비님이 설정해주신 service exception 상속받아서 던지는걸로 수정했습니다

hikarigin99 reviewed (commented):
첫 PR!! 감격입니다😭 답변해주시면 Merge 하겠습니다!

![image](https://github.com/user-attachments/assets/29ee3e1b-be05-43cf-a4fa-4be03e788e82)

w3schools-test committed:
chore: User엔티티의 setter 삭제
UserDTO 클래스 명 변경 - dto.UserDTO -> dto.request.UserReq
Error code에 EAMIL_ALREADY_EXIST 추가
Join Service에서 exception 사용

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/dto/converter/UserConverter.java:
```diff
@@ -1,48 +1,48 @@
 package com.swyp.mema.domain.user.dto.converter;
```
컨버터... dto 하위에 있었군여ㅜ 저희 패키지 컨벤션에서는 dto 와 같은 레벨에 있어야 합니다😭

codingbaraGo replied:
앗 그랬군요... 
감사합니다

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/exception/EmailAlreadyExistException.java:
```diff
@@ -0,0 +1,13 @@
+import com.swyp.mema.global.base.exception.ErrorCode;
+import com.swyp.mema.global.base.exception.ServiceException;
+
+public class EmailAlreadyExistException extends ServiceException {
```
👍

hikarigin99 commented on src/main/java/com/swyp/mema/domain/user/dto/oauth2/CustomOAuthUser.java:
```diff
@@ -1,6 +1,6 @@
 package com.swyp.mema.domain.user.dto.oauth2;
```
패키지 구조 dto.response -> CustomOAuthUser.java 로 가시는거 어떠신가용🤔

codingbaraGo replied:
dto.oauth2 내부에 있는 애들은 네이버 서버 <-> 메마 백엔드 서버 간의 상호작용 dto라 따로 빼놨어요
dto.response에는 프론트로 보내는거만 놔두려구요

그냥 dto.response에 넣어놓는게 나을까요?

hikarigin99 replied:
넵 저는 response 안에 넣어도 좋을 것 같습니다!
아님 response.oauth2 패키지 만들어도 좋을 것 같아용☺️

codingbaraGo replied:
반영 완료했습니다!
response.oauth2 만들었어요

hikarigin99 reviewed (commented):
고생하셨습니다👍 소소한 패키지 경로 수정만 바로 되면 merge 할게요!

w3schools-test committed:
chore: user.dto.converter 패키지 위치 -> user.converter로 수정

w3schools-test committed:
chore: user.dto.oauth2.CustomOAuthUser 클래스 이동 -> user.dto.CustomOAuthUser
user.dto.oauth2 패키지 이동 -> user.dto.response.oauth2

hikarigin99 commented:
고생하셨습니당👏👏👏 최종 확인되어 Merge 하겠습니다!
Merge 완료되면 개발 서버에서 API 제대로 나왔는지 한번 확인해주세요😇
